daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -36,7 +45,22 @@ private[speedy] sealed abstract class SBuiltin(val arity: Int) {
   private[speedy] def execute(args: util.ArrayList[SValue], machine: Machine): Unit
 }
 
+private[speedy] sealed abstract class SBuiltinPure(val arity1: Int) extends SBuiltin(arity1) {
+
+  override private[speedy] final def execute(
```

Comment:
Would it make sense to put some sort of inlining annotation here?

Reply:
my earlier experience is that this kind of attempted micro optimization has no benefit.